ninja-standalone/src/main/java/ninja/standalone/NinjaJetty.java

```diff
@@ -90,111 +118,196 @@ public Injector getInjector() {
     }
     
     public NinjaJetty setPort(int port) {
-        
         this.port = port;
         return this;
-        
     }
     
-    public NinjaJetty setServerUri(URI serverUri) {
-        
-        this.serverUri = serverUri;
+    public NinjaJetty setHost(String host) {
+        this.host = host;
         return this;
     }
+
+    public NinjaJetty setJettyConfiguration(String jettyConfiguration) {
+        this.jettyConfiguration = jettyConfiguration;
+        return this;
+    }
+
+    public void setServerUri(URI serverUri) {
+        this.serverUri = serverUri;
+    }
+    
+    public String createServerName() {
+        String h = this.host;
+        if (this.host == null) {
+            h = "localhost";
+        }
+        return "http://" + h + ":" + this.port;
+    }
     
     public NinjaJetty setNinjaMode(NinjaMode ninjaMode) {
-        
         this.ninjaMode = ninjaMode;
         return this;
     }
 
     public NinjaJetty setNinjaContextPath(String ninjaContextPath) {
-
         this.ninjaContextPath = ninjaContextPath;
         return this;
     }
     
-    public void start() {
+    static private Server createServerOrApplyConfiguration(String jettyConfiguration, Server server) throws Exception {
+        
+        // try local file first
+        Resource jettyConfigurationFile = Resource.newResource(jettyConfiguration);
+        
+        if (jettyConfigurationFile == null || !jettyConfigurationFile.exists()) {
+            
+            // fallback to classpath
+            jettyConfigurationFile = Resource.newClassPathResource(jettyConfiguration);
+            
+            if (jettyConfigurationFile == null || !jettyConfigurationFile.exists()) {
+            
+                throw new FileNotFoundException("Unable to find jetty configuration file either locally or on classpath [" + jettyConfiguration + "]");
+                
+            }
+        }
+        
+        System.out.println("Using jetty configuration file to configure server: " + jettyConfigurationFile);
```

Comment:
Depends. Ninja's logging config is only relevant if you did not set the config via -Dlogback.configurationFile. Otherwise the whole config is skipped and logback configures itself without Ninja. We could init the configuration even further above if we get hold of ninjaproperties (which we can do if we use NinjaPropertiesImpl).